pyrfc/_pyrfc.pyx

```diff
@@ -2051,19 +2051,20 @@ cdef wrapString(SAP_UC* uc, uclen=-1, rstrip=False):
     if uclen == -1:
         uclen = strlenU(uc)
     if uclen == 0:
-        return ''
-    cdef unsigned utf8_size = uclen * 3 + 1
+        return ''
+    cdef unsigned utf8_size = uclen * 5 + 1
     cdef char *utf8 = <char*> malloc(utf8_size)
     utf8[0] = '\0'
     cdef unsigned result_len = 0
     rc = RfcSAPUCToUTF8(uc, uclen, <RFC_BYTE*> utf8, &utf8_size, &result_len, &errorInfo)
     if rc != RFC_OK:
         # raise wrapError(&errorInfo)
         raise RFCError('wrapString uclen: %u utf8_size: %u' % (uclen, utf8_size))
+    utf8[result_len] = '\0'
     try:
         if rstrip:
-            return utf8.rstrip().decode('UTF-8')
+            return utf8[:result_len].rstrip().decode('UTF-8')

```

Comment:
I am sure that this is the key fix - it was not trimming out trailing zeroes (or junk) before attempted to decode